client/data/marketplace/use-site-search-es-query.ts

```diff
@@ -0,0 +1,217 @@
+			tested: hit.tested,
+			rating: hit.rating,
+			num_ratings: hit.num_ratings,
+			support_threads: hit.support_threads,
+			support_threads_resolved: hit.support_threads_resolved,
+			active_installs: hit.active_installs,
+			last_updated: hit[ 'meta.last_updated.value' ],
+			short_description: hit.excerpt_en, // TODO: add localization
+			icons: createIconsObject( hit.slug, hit[ 'meta.assets_icons.value' ] ),
+		};
+	} );
+};
+
+/**
+ *
+ * @param url The url to post to
+ * @param body The body of the request
+ * @returns {Promise<Response>} A promise that resolves to the response body
+ */
+async function postRequest( url: string, body: object ) {
+	const response = await fetch( `${ url }`, {
+		method: 'POST',
+		body: JSON.stringify( body ),
+		headers: { Accept: 'application/json', 'Content-Type': 'application/json' },
+	} );
+
+	if ( response.ok ) {
+		return await response.json();
+	}
+	throw new Error( await response.text() );
+}
+
+export async function fetchWPOrgPluginsFromSiteSearch( options: PluginQueryOptions ) {
+	const category = options.category || DEFAULT_CATEGORY;
+	const search = options.searchTerm;
+	const author = options.author;
+
+	// Set up base query
+	const size = options.pageSize || DEFAULT_PAGE_SIZE;
+	const from = ( ( options.page || DEFAULT_FIRST_PAGE ) - 1 ) * size;
+	const must: object[] = [];
+	const should: object[] = [];
+	const query = {
+		bool: {
+			must,
+			should,
+		},
+	};
+
+	// add a match clause to the query
+	const addMatch = ( term: object[], match: object ) => {
+		term.push( { match } );
+	};
+
+	if ( search ) {
+		must.push( {
+			multi_match: {
+				fields: [
+					'title.en^0.1',
+					'content.en^0.1',
+					'excerpt.en^0.1',
+					'tag.name.en^0.1',
+					'category.name.en^0.1',
+					'author_login^0.1',
+					'author^0.1',
+					'slug^0.1',
+				],
+				query: search,
+				operator: 'and',
+			},
+		} );
+	}
+
+	if ( author ) {
+		addMatch( must, { author } );
+	}
+
+	// matching category if no other criteria given
+	if ( ! search && ! author ) {
+		addMatch( must, { 'taxonomy.plugin_category.name': category } );
+	}
+
+	const fields = [
+		'rating',
+		'post_id',
+		'meta.assets_icons.value',
+		'meta.header_author.value',
+		'meta.header_author_uri.value',
+		'meta.last_updated.value',
+		'taxonomy.plugin_category.name',
+		'taxonomy.plugin_tags.name',
+		'excerpt_en',
+		'title_en',
+		'stable_tag',
+		'author',
+		'tested',
+		'num_ratings',
+		'support_threads',
+		'support_threads_resolved',
+		'active_installs',
+		'last_updated',
+		'slug',
+	];
+
+	const requestResult = await postRequest( SITE_SEARCH_ENDPOINT, {
+		query,
+		fields,
+		size,
+		from,
+	} );
+
+	return requestResult.results;
+}
+
+const extractPages = ( pages: Array< { hits: ESHits } > = [] ) => {
+	const allHits = pages.flatMap( ( page ) => page.hits );
+	return normalizePluginsList( mapIndexResultsToPluginData( allHits ) );
```

Comment:
Nitpicking: I think we can simplify this piece in the future. We are mapping the ES results to look like the old requests and then we normalize this data for usage in the application. I would say we could do this straight from ES results to application-ready data.

That said, I think it should be a follow-up PR as we are changing a lot already at this one.

Reply:
I agree that the mapping could be further simplified, but I wanted to touch as few layers as possible with the MVP. My original idea was to replace only the fetch layer within the react-query and do a data-mapping that would let the `select` pick up the response the same way. Unfortunately, the responses are so different (e.g pagination for the ES response has to be calculated from the query params and the `total` number in the response), that I had to create a new query after all.

Later, when all results can be queried from the same index, we can definitely come up with a simpler mapping and remove a lot of redundant logic.